lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -2672,3 +2845,42 @@ func (d *zfs) RenameVolumeSnapshot(vol Volume, newSnapshotName string, op *opera
 
 	return nil
 }
+
+// FillVolumeConfig populate volume with default config.
+func (d *zfs) FillVolumeConfig(vol Volume) error {
+	// Copy volume.* configuration options from pool.
+	err := d.fillVolumeConfig(&vol)
+	if err != nil {
+		return err
+	}
+
+	// Only validate filesystem config keys for filesystem volumes or VM block volumes (which have an
+	// associated filesystem volume).
+	if d.isBlockBacking(vol) && vol.ContentType() == ContentTypeFS || vol.IsVMBlock() {
```

Comment:
This got me thinking, are we going to use block backed volumes (if enabled on the pool) for VM config volumes?
Was this discussed/considered in the specification in terms of pros/cons?
As the VM config volume is an internal concept, not exposed to users really, I wonder if the zfs.block_backed setting should influence them at all, and we should just stick with a normal ZFS dataset for those?

What do you think @monstermunchkin @stgraber ?

Reply:
@stgraber confirmed that "Those should remain ZFS datasets".

Reply:
@monstermunchkin did this change get made?

Reply:
@monstermunchkin any news on this?